### PR TITLE
Bessel and Related Functions

### PR DESCRIPTION
**Pull request by 0x00b1**
Adds:

```Python
airy_ai(input, *, out=None) -> Tensor
```

Airy function $\text{Ai}(\text{input})$.

```Python
airy_bi(input, *, out=None) -> Tensor
```

Airy function $\text{Bi}(\text{input})$.

```Python
airy_derivative_ai(input, *, out=None) -> Tensor
```

Derivative of Airy function $\text{Ai}(\text{input})$.

```Python
airy_derivative_bi(input, *, out=None) -> Tensor
```

Derivative of Airy function $\text{Bi}(\text{input})$.

```Python
scaled_modified_bessel_i0(input, *, out=None) -> Tensor
```

Scaled modified Bessel function of the first kind of order $0$, $\text{exp}(\text{input})\text{I}_{0}(\text{input})$.

```Python
scaled_modified_bessel_i1(input, *, out=None) -> Tensor
```

Scaled modified Bessel function of the first kind of order $1$, $\text{exp}(\text{input})\text{I}_{1}(\text{input})$.

```Python
scaled_modified_bessel_k0(input, *, out=None) -> Tensor
```

Scaled modified Bessel function of the second kind of order $0$, $\text{exp}(\text{input})\text{K}_{0}(\text{input})$.

```Python
scaled_modified_bessel_k1(input, *, out=None) -> Tensor
```

Scaled modified Bessel function of the second kind of order $1$, $\text{exp}(\text{input})\text{K}_{1}(\text{input})$.

```Python
spherical_bessel_j0(input, *, out=None) -> Tensor
```

Spherical Bessel function of the first kind of order $0$, $\text{j}_{0}(\text{input})$.

```Python
spherical_bessel_j1(input, *, out=None) -> Tensor
```

Spherical Bessel function of the first kind of order $1$, $\text{j}_{1}(\text{input})$.

```Python
spherical_bessel_y0(input, *, out=None) -> Tensor
```

Spherical Bessel function of the first kind of order $0$, $\text{y}_{0}(\text{input})$.

```Python
spherical_bessel_y1(input, *, out=None) -> Tensor
```

Spherical Bessel function of the first kind of order $1$, $\text{y}_{1}(\text{input})$.

### CODE REVIEW

File: test/allowlist_for_publicAPI.json [hunk 1]
```diff
@@ -2956,6 +2956,10 @@
   ],
   "torch.special": [
     "airy_ai",
+    "bessel_j0",
```
Comment (albanD) [inline test/allowlist_for_publicAPI.json:2959]: I think I missed it in other PRs. But we should not add anything new to this file.
We should properly cleanup the special submodule instead.

Comment (0x00b1) [reply test/allowlist_for_publicAPI.json:2959]: @albanD Is this something I can do? What needs to be done?

Comment (albanD) [reply test/allowlist_for_publicAPI.json:2962]: You can check the error message when you don't add this for details.
But most likely adding a `__all__` to the special module will fix this.

Comment (0x00b1) [reply test/allowlist_for_publicAPI.json:2962]: The special module already defines `__all__`.

**Pull request closed without merging**